### ISSUE TITLE
Autocomplete emit onClear event only once completeMethod event has been triggered (with forceSelection)

### ISSUE THREAD

**Issue by florentL** [primefaces/primeng#5474]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Plunkr Case (Bug Reports)**

http://plnkr.co/edit/YEb5oLuiBEPFqugmWHxV?p=preview (I copied the plunkr link of someone else using the v5.2.0 but this behaviour is the same for v5.2.4)

**Current behavior**
Autocomplete with forceSelection option set on true emit onClear event only once completeMethod event has been triggered.

**Expected behavior**
OnClear event should emit when value is cleared because of forceSelection option.

**Minimal reproduction of the problem with instructions**
Scenario in the exemplary plunkr:

1. Open console (F12) to be able to observe console.log.
2. Choose Volvo.
3. Delete last 'o' from 'Volvo' to have 'Volv' then hit the tab button on your keyboard immediately
5. Value is not cleared.
6. There is no console.log. (onClear event hasn't emitted)

**What is the motivation / use case for changing the behavior?**
To be able to watch for empty values everytime.

* **Angular version:** 5.2.0

* **PrimeNG version:** 5.2.4

* **Browser:** [all]
 
* **Language:** [all]

**Comment by cagataycivici** [comment]
Should be fixed recently, please create a new ticket if the issue persists.
